Perl/shared/snappy/csnappy_decompress.c

```diff
@@ -371,7 +371,7 @@ csnappy_decompress_noheader(
 				src += extra_bytes;
 				available = end_minus5 + 5 - src;
 			}
-                        if (unlikely(available < (int32_t)length))
+                        if (unlikely(available < (uint32_t)length))
```

Comment:
`length` already is a `uint32_t`; perhaps the cast should be on `available` instead?